sycl/doc/extensions/experimental/sycl_ext_oneapi_annotated_ptr.asciidoc

```diff
@@ -284,7 +284,6 @@ class annotated_ptr {
     explicit operator bool() const noexcept;
 
     // Implicit conversion is not supported
-    operator T*() noexcept = delete;
```

Comment:
The const version can be called in all cases and therefore this will always produce the diagnostic that it is explicitly deleted. Note that we wouldn't need to delete either to prevent implicit conversion - the lack of conversion would be sufficient. It just gives better diagnostic to delete it.